File: 2008/D.py
import math

movimentos = 2

while True:
    x1 = input()
    y1 = input()
    x2 = input()
    y2 = input()
    
    if int(x1) == 0 and int(x2) == 0 and int(y1) == 0 and int(y2) == 0:
        break
    
    
    vertical = (int(x1) == int(x2) and int(y1) != int(y2))
    horizontal = (int(x1) != int(x2) and int(y1) == int(y2))
    diagonal = (math.fabs(int(x1)-int(x2)) == math.fabs(int(y1)-int(y2)))
    
    if vertical or horizontal or diagonal:
        movimentos = 1
    
    print(movimentos)

File: 2008/D.cpp
#include<iostream>
#include <stdbool.h>
#include <stdlib.h> 
using namespace std;

int main(){
	int x1, x2, y1, y2, movimentos = 2;
	while(true){
		movimentos = 2;
		cin >> x1 >> y1 >> x2 >> y2;
		
		if(x1 == 0 && x2 == 0 && y1 == 0 && y2 == 0){
			break;
		}
		
		bool vertical = (x1 == x2 && y1 != y2);
		bool horizontal = (x1 != x2 && y1 == y2);
		bool diagonal = (abs(x1-x2) == abs(y1-y2));
		
		if(x1 == x2 && y1 == y2){
			movimentos = 0;
		}else if(vertical || horizontal || diagonal){
			movimentos = 1;
		}
		cout << movimentos << endl;
	}
}
